contracts/USM.sol

```diff
@@ -459,17 +457,21 @@ contract USM is IUSM, Oracle, ERC20WithOptOut, Delegable {
     // ____________________ Public helper view functions (for functions above) ____________________
 
     function loadState() public view returns (LoadedState memory ls) {
-        ls.timeSystemWentUnderwater = storedState.timeSystemWentUnderwater;
-        ls.ethUsdPriceTimestamp = storedState.ethUsdPriceTimestamp;
-        ls.ethUsdPrice = storedState.ethUsdPrice * BILLION;     // Converting stored BILLION (10**9) format to WAD (10**18)
-        ls.buySellAdjustmentTimestamp = storedState.buySellAdjustmentTimestamp;
-        ls.buySellAdjustment = storedState.buySellAdjustment * BILLION;
-        ls.ethPool = ethPool();
-        ls.usmTotalSupply = totalSupply();
-
-        // Bring buySellAdjustment from its stored time to the present - it gravitates towards 1 over time:
-        ls.buySellAdjustment = buySellAdjustment(ls.buySellAdjustmentTimestamp, ls.buySellAdjustment, block.timestamp);
-        ls.buySellAdjustmentTimestamp = block.timestamp;
+        (ls.timeSystemWentUnderwater,
```

Comment:
I think that you can write this in a simpler manner like:
```
ls = LoadedState({
    storedState.timeSystemWentUnderwater,
    storedState.ethUsdPriceTimestamp,
    storedState.ethUsdPrice * BILLION,     // Converting stored BILLION (10**9) format to WAD (10**18)
    block.timestamp,   // Bring buySellAdjustment from its stored time to the present: it gravitates -> 1 over time
    buySellAdjustment(
        storedState.buySellAdjustmentTimestamp,
        storedState.buySellAdjustment * BILLION,
        block.timestamp
    ),
    ethPool(),
    totalSupply()
});
```

Reply:
Yeah there are a bunch of ways to load and store a struct and I was just trying to do the ones that gave the best gas results.  I redid my tests and here's what I got, so I'm just going to go with the cheapest options - they may not always be cheapest, or sometimes it may not matter, but I just don't want to have to think about it every time!  So I think these are reasonable default approaches to use.

I could swear some approaches were more inefficient than shown here (hundreds of gas extra), but I can't reproduce that right now.

Loading:

Code | Gas | Gas Diff
--- | --- | ---
`ls.k1 = v1; ls.k2 = v2; ...` | 115,898 gas | +0
`(ls.k1, ls.k2, ...) = (v1, v2, ...)` | 115,928 gas | +30
`ls = LoadedState({k1: v1, k2: v2, ...})` | 115,966 gas | +68
`ls = LoadedState(v1, v2, ...)` | 115,966 gas | +68

Storing:

| Code | Gas | Gas Diff |
| --- | --- | --- |
| `(storedState.k1, storedState.k2, ...) = (v1, v2, ...)` | 115,898 gas | +0 |
| `storedState.k1 = v1; storedState.k2 = v2; ...` | 115,913 gas | +15 |
| `storedState = StoredState({k1: v1, k2: v2, ...})` | 116,036 gas | +138 |
| `storedState = StoredState(v1, v2, ...)` | 116,036 gas | +138 |
| `StoredState memory ss; ss.k1 = v1; ss.k2 = v2; ...; storedState = ss` | 116,156 gas | +258 |

Reply:
Very interesting!